packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -169,6 +171,56 @@ impl Resolver {
         }
     }
 
+    /// Delete `nodes` from the resolver
+    pub fn delete_nodes(&mut self, nodes: &[FlatNode]) {
+        for node in nodes.iter() {
+            if let FlatNode::Element(element) = node {
+                if let Some(name) = get_element_name(element) {
+                    let mut parent_option = self.node_parent[node.idx()];
+
+                    while let Some(parent_idx) = parent_option {
```

Comment:
I think this could be written better by having `parent_idx` be the mutable thing and then directly doing a while loop over `self.node_parent[parent_idx]`